Added ColorMapExpression Spec

ggetz opened the pull request:
* Added ColorMapExpression Spec
   * Is the file long enough that I should bring `jsonExp` into the specs for readability?
* Added check in setter for `default` so it does not mark styleEngine as dirty if the value doesn't change

pjcozzi commented on Source/Scene/ColorMapExpression.js:
```diff
@@ -74,10 +74,12 @@ define([
             set : function(value) {
-                this._default = value.slice(0);
-                this._styleEngine.makeDirty();
+                if (this._default !== value) {
```
This is a reference compare; instead, we would want to check each element in the array.

However, I actually want to change how we expose `map` and `default`.  Let's expose the "compiled" version, which uses `Color` like we do elsewhere (we'll want to do the same thing for color ramp), not the style JSON version.

pjcozzi commented on Specs/Scene/ColorMapExpressionSpec.js:
```diff
@@ -0,0 +1,121 @@
+    MockStyleEngine.prototype.makeDirty = function() {
+    };
+
+    function MockFeature(key) {
```
In the generic context of a `Feature`, `key` should probably be `value` since it isn't always a key.

pjcozzi commented:
> Is the file long enough that I should bring jsonExp into the specs for readability?

How you have it is good.

pjcozzi commented:
@ggetz let me know when this is ready.

ggetz commented:
Ready.

pjcozzi commented on Specs/Scene/ColorMapExpressionSpec.js:
```diff
@@ -0,0 +1,121 @@
+
+    it('sets map', function() {
+        var expression = new ColorMapExpression(new MockStyleEngine(), jsonExp);
+        expression.map = {
```
This should use the "compiled" version with Cesium's `Color` type.

ggetz commented:
I updated the setters so they now used the Cesium Color objects. I removed the call to `setRuntime` in the setters as well, because they are no longer needed and override the run time colors with the initial values.
